client/app/containers/App/epic.js

```diff
@@ -1,15 +1,48 @@
+import { Observable } from 'rxjs';
 import { LOCATION_CHANGE } from 'react-router-redux';
 import {
   closeDrawer,
 } from './actions';
+import {
+  changeToPage,
+  getPageOfPosts,
+} from '../Blog/actions';
 
 // Automatically close the nav drawer when the user has selected a new route
-// TODO: Filter this so it only catches when the drawer is active, currently
-// is a brute force, always fire action...
-const closeNavEpic = (action$) =>
+const closeNavEpic = (action$, store) =>
+  action$.ofType(LOCATION_CHANGE)
+    .switchMap(() => {
+      const drawerActive = store.getState().getIn(['global', 'drawerActive']);
+      if (drawerActive) {
+        return Observable.of(closeDrawer());
+      }
+      return Observable.of();
+    });
+
+// Intelligently use state to determine if we need to fetch the requested page,
+// or can simply use cached content. Also provides support for if we change pages
+// using react-router navigation instead of actions.
+const fetchPageEpic = (action$, store) =>
   action$.ofType(LOCATION_CHANGE)
-    .map(() => closeDrawer());
+    .switchMap((action) => {
+      // If location change is to blog pages, parse out the page and fetch that data
```

Comment:
Mhmm, looking at this it seemed like a good way to reduce having a bunch of possibly colliding async results: http://stackoverflow.com/a/42227335/7027882